rephrase_chempile/rephrase.py

```diff
@@ -0,0 +1,349 @@
+    )
+    messages: list[Message]
+
+
+CONVERSATION_SCHEMA = Conversation.model_json_schema()
+
+HARD_PROMPTS = [
+    "Use esoteric vocabulary and complex syntax suitable for academics",
+    "Employ arcane terminology and intricate sentence structures for scholarly audiences",
+    "Utilize obscure language and convoluted phrasing to challenge comprehension",
+    "Incorporate recondite expressions and sophisticated grammar for expert readers",
+    "Apply cryptic diction and elaborate constructions to elevate academic rigor",
+]
+
+WIKI_PROMPTS = [
+    "Use formal, encyclopedic English resembling Wikipedia",
+    "Employ objective, authoritative language similar to an encyclopedia entry",
+    "Utilize precise, factual phrasing akin to scholarly reference works",
+    "Adopt neutral, informative tone characteristic of academic articles",
+    "Present information in clear, impersonal style as found in reference materials",
+]
+
+ENGAGING_PROMPTS = [
+    "Present information in a clear yet engaging manner, suitable for a broad audience",
+    "Use an approachable, conversational tone that still maintains a professional and informative style",
+    "Employ simple yet precise language with a touch of sophistication, making it both accessible and authoritative",
+    "Balance clarity with expert-level knowledge, presenting facts in a digestible and friendly way",
+    "Utilize an engaging, direct style that appeals to a wide range of readers while retaining intellectual rigor",
+]
+
+
+def prepare_jsonl_file(df: pd.DataFrame, dset: str):
+    import random
+
+    tasks = []
+    prompt_sets = [HARD_PROMPTS, WIKI_PROMPTS, ENGAGING_PROMPTS, ""]
+    weights = [0.1, 0.3, 0.3, 0.3]
+    for i, row in df.iterrows():
+        text = row["text"]
+        chosen_set = random.choices(prompt_sets, weights=weights, k=1)[0]
+        if isinstance(chosen_set, list):
+            user_prompt = random.choice(chosen_set)
+        else:
+            user_prompt = ""
+        prompt_text = f"{user_prompt}. " if user_prompt else ""
+        task = {
+            "custom_id": f"task-{dset}-{i}",
+            "method": "POST",
+            "url": "/v1/chat/completions",
+            "body": {
+                "model": "gpt-4o-mini-2024-07-18",
+                "messages": [
+                    {"role": "system", "content": SYSTEM_PROMPT},
+                    {
+                        "role": "user",
+                        "content": f"{prompt_text}\nOriginal text:\n{text}\n\nNow, rephrase this text into a multi-turn conversation about chemistry.",
+                    },
+                ],
+                "response_format": {
+                    "type": "json_schema",
+                    "json_schema": {
+                        "name": "multi_turn_conversation",
+                        "schema": CONVERSATION_SCHEMA,
+                    },
+                },
+            },
+        }
+        tasks.append(task)
+    f_name = f"{dset}.jsonl"
```

Comment:
I wonder if you should also store somewhere what prompt style was used for rephrasing. This might be useful at some point. And storing might be easier than reverse-engineering it later.